ydb/core/blobstorage/vdisk/balance/balancing_actor.cpp

```diff
@@ -1,29 +1,191 @@
+                Send(Ctx->SkeletonId, new TEvStartBalancing());
+                Send(SelfId(), new NActors::TEvents::TEvPoison);
+            }
+        }
+
+        void Handle(TEvProxyQueueState::TPtr ev) {
+            const TVDiskID& vdiskId = ev->Get()->VDiskId;
+            if (ev->Get()->IsConnected) {
+                ConnectedVDisks.insert(vdiskId);
+            } else {
+                ConnectedVDisks.erase(vdiskId);
+            }
+        }
+
+        void Handle(TEvVGenerationChange::TPtr ev) {
+            Cerr << "TEvVGenerationChange" << Endl;
+            // forward message to queue actors
+            TEvVGenerationChange *msg = ev->Get();
+            for (const auto& kv : *QueueActorMapPtr) {
+                Send(kv.second, msg->Clone());
+            }
+            Ctx->GInfo = msg->NewInfo;
+        }
+
+        void DoJobQuant(const TActorContext& ctx) {
+            LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                Ctx->GInfo->GetTopology().GetOrderNumber(Ctx->VCtx->ShortSelfVDisk) << "$ " << "Connected vdisks " << ConnectedVDisks.size() << "/" << Ctx->GInfo->GetTotalVDisksNum() - 1);
+            Send(SenderId, new NActors::TEvents::TEvWakeup());
+            Send(DeleterId, new NActors::TEvents::TEvWakeup());
+        }
+
+        void PassAway() override {
+            for (const auto& kv : *QueueActorMapPtr) {
+                Send(kv.second, new TEvents::TEvPoison);
+            }
+            TActorBootstrapped::PassAway();
         }
 
         STRICT_STFUNC(StateFunc,
-            CFunc(NActors::TEvents::TEvPoison::EventType, Die)
+            CFunc(TEvReplToken::EventType, HandleReplToken)
+            HFunc(NActors::TEvents::TEvCompleted, Handle)
+            hFunc(TEvProxyQueueState, Handle)
+            hFunc(TEvVGenerationChange, Handle)
+            CFunc(NActors::TEvents::TEvWakeup::EventType, StartBalancing)
+            cFunc(NActors::TEvents::TEvPoison::EventType, PassAway)
         );
 
+    public:
         TBalancingActor(std::shared_ptr<TBalancingCtx> &ctx)
             : TActorBootstrapped<TBalancingActor>()
             , Ctx(ctx)
+            , It(Ctx->Snap.HullCtx, &Ctx->Snap.LogoBlobsSnap)
         {
         }
+
+        void Bootstrap(const TActorContext &ctx) {
+            CreateVDisksQueues();
+            auto [sendOnMainParts, tryDeleteParts] = CollectKeys(ctx);
+            LOG_DEBUG_S(ctx, NKikimrServices::BS_VDISK_BALANCING,
+                Ctx->GInfo->GetTopology().GetOrderNumber(Ctx->VCtx->ShortSelfVDisk) << "$ "
+                << ctx.Now().MilliSeconds() << " Bootstrap" << ": "
+                << "sendOnMainParts size = " << sendOnMainParts.size() << "; tryDeleteParts size = " << tryDeleteParts.size());
+
+            SenderId = ctx.Register(CreateSenderActor(SelfId(), std::move(sendOnMainParts), QueueActorMapPtr, Ctx));
+            ActiveActors.Insert(SenderId, __FILE__, __LINE__, ctx, NKikimrServices::BLOBSTORAGE);
```

Comment:
you add actor here and take no operation when it dies and on PassAway
- on PassAway you should fan-out TEvPosion message to currently active actors
- when child actor dies, it should notify parent actor somehow (probably in TEvCompleted)